crates/holochain/tests/tests/websocket/mod.rs

```diff
@@ -640,6 +640,22 @@ async fn connection_limit_is_respected() {
     // Disconnect all the clients
     handles.clear();
```

Comment:
The test is relying on this closing connections quickly enough for the next step to open connections up to the limit again. That's a race condition because that's down to how quickly the `Drop` happens and Holochain finds out those connections are gone.

I've added a bit of code here that waits until at least one connection can be made. The idea being that once one connection is dropped, the rest should be close behind and each connection will take a bit of time to establish. 

If this fix doesn't work then I need to rethink the next step in the test.